src/test/java/org/broadinstitute/hellbender/utils/IntervalUtilsUnitTest.java

```diff
@@ -439,6 +440,149 @@ public void testParseIntervalArguments() {
+        final SAMSequenceDictionary sd = getAmbiguousSequenceDictionary();
+        sd.addSequence(new SAMSequenceRecord("contig:1-10", 99));
+        sd.addSequence(new SAMSequenceRecord("contig", 99));
+        SimpleInterval.getResolvedIntervals("contig:abc-dce", sd);
+    }
+
+    @Test
+    public void testMalformedAmbiguousQuerySucceeds() {
+        final SAMSequenceDictionary sd = getAmbiguousSequenceDictionary();
+        final String queryString = "contig:abc-dce";
+
+        sd.addSequence(new SAMSequenceRecord("contig", 99));
+        sd.addSequence(new SAMSequenceRecord(queryString, 99));
+
+        // the query string matches as a full contig, but also has a prefix that matches the contig "contig". When
+        // viewed as the latter, however, the start-end positions fail to parse as numbers. This resolves as a query
+        // against the longer contig "contig:abc-dce", and logs a warning
+        HashSet<SimpleInterval> expectedIntervals = new HashSet<>();
+        expectedIntervals.add(new SimpleInterval(queryString, 1, 99));
+        Assert.assertEquals(
+                new HashSet<>(SimpleInterval.getResolvedIntervals(queryString, sd)),
+                new HashSet<>(expectedIntervals));
+    }
+
+    @Test
+    public void testUseAllHG38ContigsInIntervalQuery() {
+        SAMSequenceDictionary sd;
+
+        final File testFile = new File ("src/test/resources/org/broadinstitute/hellbender/engine/Homo_sapiens_assembly38.headerOnly.vcf.gz");
+        try (VCFFileReader vcfReader = new VCFFileReader(testFile, false)) {
+            sd = vcfReader.getFileHeader().getSequenceDictionary();
+        }
+
+        // Pull each contig from hg38 and validate that it can be used as a interval string without
+        // being rejected as ambiguous.
+        sd.getSequences().stream().forEach(
+                hg38Contig -> {
+                    // query using just the raw contig name
+                    List<SimpleInterval> ambiguousIntervals = SimpleInterval.getResolvedIntervals(hg38Contig.getSequenceName(), sd);
+                    Assert.assertNotNull(ambiguousIntervals);
+                    Assert.assertEquals(ambiguousIntervals.size(), 1);
+
+                    // query using an interval with the format "name:1-end"
+                    ambiguousIntervals = SimpleInterval.getResolvedIntervals(
+                            new SimpleInterval(hg38Contig.getSequenceName(), 1, hg38Contig.getSequenceLength()).toString(), sd);
+                    Assert.assertNotNull(ambiguousIntervals);
+                    Assert.assertEquals(ambiguousIntervals.size(), 1);
+
+                    // query using an interval with the format "name:1+"
+                    ambiguousIntervals = SimpleInterval.getResolvedIntervals(hg38Contig.getSequenceName() + ":1+", sd);
+                    Assert.assertNotNull(ambiguousIntervals);
+                    Assert.assertEquals(ambiguousIntervals.size(), 1);
+                }
```

Comment:
Also test the ":1" case, and check the contents of `ambiguousIntervals` to make sure the interval it contains is correct.

Reply:
Done.